feature(pro_sources): craftctl chroot cmd (#1350)
canonical/craft-parts: docs/reference/changelog.rst

- Validate that filesystem mounts are ordered in increasing ``mount`` nesting.
  A ``mount`` value cannot be a parent of any preceding ``mount`` values.

> - Introduced a new sub command called ``chroot`` to ``craftctl`` which allows users to
>   run a command inside the overlay of a part.

Reviewer: ```suggestion
- Add ``craftctl chroot`` sub-command, for running a command inside a part's overlay.
```
